### DISCUSSION TITLE
2 spike hubs connected to 1 xbox controller

### DISCUSSION THREAD

**Discussion by Paralysis-03** [pybricks/support#1884] [Q&A]
I have run out of ports on one of my spike prime hub and have had to resort to using a second hub. I can control one hub from an Xbox controller but not both. How do I connect 1 xbox controller to 2 Spike hubs?

**Comment by BertLindeman** [comment]
Hello,

Maybe the an example can help: in discussions Show and tell:  
[Liebherr R 9800 Excavator 42100 with Xbox controller](https://github.com/orgs/pybricks/discussions/1738)

Bert

**Comment by laurensvalk** [comment]
[Here is an example](https://pybricks.com/project/technic-42146-xbox/) of controlling two hubs with one Xbox Controller. The same technique works for SPIKE Prime.

[![image](https://github.com/user-attachments/assets/752fad4a-c18d-457d-8088-a2517db34173)](https://pybricks.com/project/technic-42146-xbox/)

**Reply by Paralysis-03** [reply]
Would it work with 1 Spike essential and 1 spike prime?

**Reply by laurensvalk** [reply]
Yes. With Pybricks all hubs work the same. I would recommend following [this tutorial](https://www.youtube.com/watch?v=WzmcihSV2YE) as well. Coincidentally, it uses the SPIKE Essential Hub and a Robot Inventor Hub (which is just a green SPIKE Prime) so this is exactly your setup.

Once you're familiar with this hub communication technique, add the Xbox Controller as in the crane example.

**Comment by Paralysis-03** [comment]
I keep getting an error:
```
Traceback (most recent call last):
  File "essential.py", line 18, in <module>
TypeError: can't convert tuple to int
```

how do i fix it?

This is my code for the reciving hub:
```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor,
from pybricks.parameters import Axis, Button, Color, Direction, Port, Side, Stop
from pybricks.tools import wait, 

hub = PrimeHub(top_side=Axis.Z, front_side=-Axis.Y, broadcast_channel=0, observe_channels=[5])
claw = Motor(Port.A, Direction.CLOCKWISE)

# Initialize variables.

claw_speed = 0


# The main program starts here.
while True:
    claw_speed = hub.ble.observe(5) or [0] * 0
    if claw_speed:
        claw.run(claw_speed * 8)
    else:
        claw.stop()
    wait(100)
```

**Reply by dlech** [reply]
If you are broadcast more than one value, you need to unpack the tuple returned by the `observe()` method.

So, you can either only broadcast only one value or change `claw_speed = hub.ble.observe(5)` to `claw_speed, *_ = hub.ble.observe(5)`
